### PR TITLE
[GH-23555] Do not block global backspace key press for contenteditable's

### PR DESCRIPTION
**Pull request by akaravashkin**
#### Summary
It allows to use `Backspace` in `<div contenteditable="true">`

I used solution from here https://github.com/mattermost/mattermost-webapp/pull/11956/files#diff-cd6ea218d93d3408816b90564beb01b90338fbde9049c3ef2f9574c696eea192R208-R212

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/23555
Jira https://mattermost.atlassian.net/browse/MM-53063

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/logged_in/logged_in.tsx [hunk 1]
```diff
@@ -206,7 +206,11 @@ export default class LoggedIn extends React.PureComponent<Props> {
     private handleBackSpace = (e: KeyboardEvent): void => {
         const excludedElements = ['input', 'textarea'];
 
-        if (e.which === BACKSPACE_CHAR && !(excludedElements.includes((e.target as HTMLElement).tagName.toLowerCase()))) {
+        if (
+            e.which === BACKSPACE_CHAR && e.target &&
+            !(excludedElements.includes((e.target as HTMLElement).tagName.toLowerCase())) &&
+            (e.target as HTMLElement).getAttribute?.('contenteditable') !== 'true'
+        ) {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/logged_in/logged_in.tsx:213]: would you be open in improving this code, this piece was written many years ago. We would want to do the following here to improve the readability 
- remove `which` its deprecated and replace it with `isKeyPressed(e, Constants.KeyCodes.BACKSPACE)` these are utility used in codebase
- create constants for `targetElement` and `targetsTagName`
- create `isTargetNotContentEditable` constant

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by hmhealey — APPROVED** [review]
Thanks for the PR! This looks great, and I'm sure it'll be helpful in the future whenever we next make improvements to the post textbox.

**Comment by akaravashkin** [comment]
@M-ZubairAhmed I tried to use "Sync fork" button in my github fork instead of rebasing manually. It is good now?

**Comment by M-ZubairAhmed** [comment]
@yasserfaraazkhan this as well we might have to create upstream branch from this fork for e2e

**Review by yasserfaraazkhan — APPROVED** [review]

**Comment by hmhealey** [comment]
Thanks again! This should be available in the July release of MM

**Pull request merged**